szar/clustering.py

```diff
@@ -146,14 +169,38 @@ def ps_bar(self,mu,fsky):
         #    ans[:,:,i] = self.HMF.dVdz[i]*nbar[i]**2*ps_tilde[:,:,i]*np.diff(z_arr[i])/self.Norm_Sfunc(fsky)[i]
         return ans
 
+    #NOT READY FOR USE
+    def ps_bar_fine(self,mu,fsky):
+        zs = self.HMF.zarr
+        ntils = self.ntilde()
+        
+        for i in range(zs.size):
+            if i == 0:
+                fine_zs = np.linspace(zs[i], zgridedges[i+1], nsubsamples)
+            elif i == zs.size - 1:
+                fine_zs = np.linspace(zgridedges[i], zs[i])
+            else:
+                fine_zs = np.linspace(zgridedges[i], zgridedges[i+1], nsubsamples)
+            ntils = self.ntilde_interpol(fine_zs)
+            dvdz = self.dVdz_fine(fine_zs)
+            prefac = dvdz * ntils**2
+            
+
+            integral = simps(dvdz * ntils**2, fine_zs)
+            values[i] = integral * 4 * np.pi * fsky
+        return values
+
     def V_eff(self,mu,fsky):
+        V0 = self.v0(fsky)
+        V0 = np.reshape(V0, (V0.size,1))
 
-        V0 = self.HMF.dVdz*np.diff(z_arr)*4.*np.pi*fsky #FIX
         nbar = self.ntilde()
+        nbar = np.reshape(nbar, (nbar.size,1))
+
         ps = self.ps_bar(mu,fsky)
-        npfact = np.multiply(ps,nbar)
-        frac = old_div(npfact, (1. + npfact))
-        ans = np.multiply(frac,V0)
+        npfact = np.multiply(nbar, ps)
+        frac = npfact/(1. + npfact)
```

Comment:
There's a typo in the paper, this need to be squared

Reply:
Weird. I thought when I pushed my commit to my master it would update that. Will fix.